vault/identity_store_oidc.go

```diff
@@ -1575,24 +1617,35 @@ func (i *IdentityStore) expireOIDCPublicKeys(ctx context.Context, s logical.Stor
 	return nextExpiration, nil
 }
 
-func (i *IdentityStore) oidcKeyRotation(ctx context.Context, s logical.Storage) (time.Time, error) {
+func (i *IdentityStore) oidcKeyRotation(ctx context.Context, s logical.Storage) (nextRotation time.Time, maxJwksClientCacheDuration time.Duration, err error) {
 	// soonestRotation will be the soonest rotation time of all keys. Initialize
 	// here to a relatively distant time.
 	now := time.Now()
 	soonestRotation := now.Add(24 * time.Hour)
 
+	// the OIDC JWKS endpoint returns a Cache-Control HTTP header time
+	// between 0 and the minimum verificationTTL or minimum rotationPeriod out
+	// of all keys, whichever value is lower.
+	//
+	// This smooths calls from services validating JWTs to Vault, while
+	// ensuring that operators can assert that servers honoring the Cache-Control
+	// header will always have a superset of all valid keys, and not trust
+	// any keys longer than a jwksCacheControlMax duration after a key is rotated
```

Comment:
nit: It seems `jwksCacheControlMax` is referring to a variable somewhere, but I couldn't find it. Suggest to update comment.